app/assets/javascripts/upload.coffee

```diff
@@ -373,60 +271,162 @@ bulkCorrectionUpload = (fileInput) ->
+  progressBarElement
+  successCallback
+  fileChangeCallBack
+  hiddenInputElement
+  single
+  ) ->
+    # update helpdesk
+    $('[data-toggle="popover"]').popover()
+    hiddenInput = document.getElementById(hiddenInputElement)
+    hiddenInput2 = document.getElementById('upload-userManuscript-hidden2')
+    fileInput =document.getElementById(fileInputElement)
+    fI = $("#"+fileInputElement)
+    fileInput.style.display = 'none'
+    result = undefined
+    merged = undefined
+    files = []
+    filez= []
+    
+
+    uploadedFiles = []
+    progressOptimize = 0
+    $(uploadStatusElement).hide()
+    uploadButton = $(uploadButtonElement)
+    $(actualUploadButtonElement).on 'click', (e) ->
+      e.preventDefault()
+      if permissionElement == null || $(permissionElement).is(":checked")
+        #Upload blob
+        i=0
+        xhr = new XMLHttpRequest()
+        onerror = (e) ->
+          console.log(e)
+          alert(
+            $(progressBarElement).data('tr-failure')
+          )
+        onprogress = (e) ->
+            percentUpload = Math.floor(100/fileInput.files.length * e.loaded / e.total+100*(i)/fileInput.files.length)
+            $(progressBarElement).text(percentUpload+" %")
+            return
+        onload = (xhr) -> () ->
+          console.log xhr.status
+          if (xhr.status == 200)
+            i++
+            uploadedFiles.push JSON.parse(xhr.responseText)
+            if  i == (fileInput.files.length)
+              uploadedFiles2 = uploadedFiles
+              if single
+                uploadedFiles2 = uploadedFiles[0]
+              if successCallback != undefined 
+                successCallback(xhr)
+              $(progressBarElement).text(
+                $(progressBarElement).data 'tr-success'
+              )
+              hiddenInput.value = JSON.stringify uploadedFiles2
+              fileInput.value = null
+              $(progressBarElement)
+                  .removeClass('btn-primary')
+                  .addClass 'btn-outline-secondary'
+            else
+              f = fileInput.files[i]
+              formData = new FormData()
+              formData.append("file", f, f.name)
+              xhr2 = new XMLHttpRequest()
+              xhr2.onload= onload(xhr2)
+              xhr2.onerror =onerror
+              xhr.upload.onprogress = onprogress
+              xhr2.open('POST', endpoint, true)
+              xhr2.send(formData)
+          else
+            console.log(xhr)
+            alert(
+              $(progressBarElement).data('tr-failure') + xhr.response
+            )
+        xhr.onload = onload(xhr)
+        
+        xhr.onerror = onerror
+        xhr.upload.onprogress = onprogress
+        f = fileInput.files[0]
+        formData = new FormData()
+        formData.append("file", f, f.name)
+        xhr.open('POST', endpoint, true)
+        xhr.send formData
+      else
+        alert(
+          $(progressBarElement).data 'tr-missing-consent'
+        )
+    
```

Comment:
Is there a way to add an additional info just after 100% is reached (like "upload is being processed")? With large files, there is this delay between the point when 100% is reached and the file info is being displayed. This might confuse some users.

Reply:
Done!